File: src/2025-02-24-deepclaude-rust/javascript/exercises/practice/phone-number/phone-number.js
//
// This is only a SKELETON file for the 'Phone Number' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const clean = (number) => {
  if (/[a-zA-Z]/.test(number)) {
    throw new Error('Letters not permitted');
  }

  const hasNonDigits = /\D/.test(number);

  let cleaned = number.replace(/\D/g, '');

  if (cleaned.length < 10) {
    if (hasNonDigits) {
      throw new Error('Punctuations not permitted');
    } else {
      throw new Error('Incorrect number of digits');
    }
  } else if (cleaned.length > 11) {
    throw new Error('More than 11 digits');
  }

  if (cleaned.length === 11) {
    if (cleaned[0] !== '1') {
      throw new Error('11 digits must start with 1');
    }
    cleaned = cleaned.substring(1);
  }

  if (cleaned[0] === '0') {
    throw new Error('Area code cannot start with zero');
  }
  if (cleaned[0] === '1') {
    throw new Error('Area code cannot start with one');
  }

  if (cleaned[3] === '0') {
    throw new Error('Exchange code cannot start with zero');
  }
  if (cleaned[3] === '1') {
    throw new Error('Exchange code cannot start with one');
  }

  return cleaned;
};

File: src/2025-02-24-deepclaude-rust/java/exercises/practice/phone-number/src/main/java/PhoneNumber.java
class PhoneNumber {
    private final String number;

    PhoneNumber(String numberString) {
        if (numberString == null || numberString.trim().isEmpty()) {
            throw new IllegalArgumentException("Phone number cannot be null or empty");
        }

        String cleaned = numberString.replaceAll("[^0-9]", "");

        if (cleaned.isEmpty()) {
            throw new IllegalArgumentException("Phone number contains no digits");
        }

        if (cleaned.length() == 11 && cleaned.charAt(0) == '1') {
            cleaned = cleaned.substring(1);
        }

        if (cleaned.length() != 10) {
            throw new IllegalArgumentException("Invalid phone number length");
        }

        if (cleaned.charAt(0) < '2' || cleaned.charAt(0) > '9') {
            throw new IllegalArgumentException("Area code must start with 2-9");
        }

        if (cleaned.charAt(3) < '2' || cleaned.charAt(3) > '9') {
            throw new IllegalArgumentException("Exchange code must start with 2-9");
        }

        this.number = cleaned;
    }

    String getNumber() {
        return number;
    }
}
